code/game/objects/items/devices/spy_sensor.dm

```diff
@@ -64,13 +64,15 @@
 /obj/item/device/spy_sensor/proc/reset()
 	if(!timer)
 		return
-	if(group.len > 3)
+	if(length(group) > 3)
 		group -= src
 		return
-	deltimer(timer)
-	for(var/obj/item/device/spy_sensor/S in group)
-		S.timer = null
-		S.group = null
+
+	if(timer && group)
```

Comment:
This timer check is redundant, there is one on line 65.
https://github.com/discordia-space/CEV-Eris/blob/360196d1233e800b7e83ce0a93d060f46d575249/code/game/objects/items/devices/spy_sensor.dm#L65-L66